examples/petsc/bps.c

```diff
@@ -401,6 +426,12 @@ int main(int argc, char **argv) {
   CeedInt P, Q;
   const CeedInt dim = 3, ncompx = 3;
   bpType bpChoice;
+  bool petscusecuda;
+  #ifdef PETSC_HAVE_CUDA
```

Comment:
Just because PETSc was compiled with CUDA support doesn't mean there is a GPU attached or that the user wishes to use it. For diagnostic purposes, I would make the memtype a run-time option and default to `VecSetType(X, VECCUDA)` when the Ceed prefers device memory. One could imagine testing (VECCUDA, memtype=host, /gpu/cuda/gen) versus (VECCUDA, memtype=device, /gpu/cuda/gen), as well as any combination of CPU and GPU on the PETSc and libCEED sides independently.

Reply:
Adjusted. If VECCUDA is supported and the backend prefers CEED_MEM_DEVICE, then I use VECCUDA. However, the user can specify what the memtype for the handoff of the pointer from PETSc to libCEED and back.